examples/instructor-gemini-flash/notebook.ipynb

```diff
@@ -0,0 +1,566 @@
+       "</g>\n",
+       "<!-- get_topic_feedback&#45;&gt;update_topics_so_far -->\n",
+       "<g id=\"edge7\" class=\"edge\">\n",
+       "<title>get_topic_feedback&#45;&gt;update_topics_so_far</title>\n",
+       "<path fill=\"none\" stroke=\"black\" stroke-dasharray=\"5,2\" d=\"M446.41,-80.86C473.08,-68.95 508.81,-53.01 537.3,-40.29\"/>\n",
+       "<polygon fill=\"black\" stroke=\"black\" points=\"538.96,-43.38 546.67,-36.11 536.11,-36.99 538.96,-43.38\"/>\n",
+       "<text text-anchor=\"middle\" x=\"594.5\" y=\"-54.8\" font-family=\"Times,serif\" font-size=\"14.00\">topic_feedback is None</text>\n",
+       "</g>\n",
+       "<!-- update_topics_so_far&#45;&gt;creator -->\n",
+       "<g id=\"edge9\" class=\"edge\">\n",
+       "<title>update_topics_so_far&#45;&gt;creator</title>\n",
+       "<path fill=\"none\" stroke=\"black\" d=\"M657.95,-36.03C713.41,-52.33 775.14,-79.93 743,-117 732.24,-129.41 710.2,-138.69 682.33,-145.61\"/>\n",
+       "<polygon fill=\"black\" stroke=\"black\" points=\"681.18,-142.29 672.23,-147.96 682.77,-149.1 681.18,-142.29\"/>\n",
+       "</g>\n",
+       "</g>\n",
+       "</svg>\n"
+      ],
+      "text/plain": [
+       "<graphviz.graphs.Digraph at 0x7f2abb0658d0>"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "app = topics_application()\n",
+    "app.visualize(\n",
+    "    output_file_path=\"statemachine\",\n",
+    "    include_conditions=True,\n",
+    "    include_state=True,\n",
+    "    format=\"png\",\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's go through the visualization:\n",
+    "1. Give the `ouline` string and the `num_required_topics` as inputs to `setup`.\n",
+    "2. `creator` will run in a loop. Generated one topic at a time.\n",
+    "3. If for whatever reason the `generated_topic` is None, this means there was some error even after some retries. In this case, `creator` will be called again.\n",
+    "4. `get_topic_feedback` will be called to get feedback on `generated_topic`.\n",
+    "5. If `topic_feedback` is None, it means that the user is satisfied. `update_topics_so_far` will be called to add the `generated_topic` to the `topics_so_far` list.\n",
+    "6. If the user is not satisfied, `creator` will be called again with the `generated_topic` and `topic_feedback` added to the `chat_history`.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "inputs = {\"outline\": \"history, science, math\", \"num_required_topics\": 3}\n",
+    "\n",
+    "while True:\n",
+    "    action, result, state = app.run(halt_after=[\"update_topics_so_far\"], inputs=inputs)\n",
+    "    # let's stop after we have the required number of topics\n",
```

Comment:
Might be nice to bake this into the graph? E.G. have a terminal node that you also halt_after that it transitions to, rather than checking state conditions in the while loop? IF that's too much this is fine for now -- thinking something like:

```python
("update_topics_so_far", "terminal", condition.expr("len(topics_so_far" == num_required_topics")),
("update_topics_so_far", "creator")
```

No worries if not though, this is a reasonable as is